src/eth/rpc.ts

```diff
@@ -30,6 +36,24 @@ const getDelegatorSnapshot = async (
     delegate,
   };
 };
+const isEOA = async (address: string) => {
+    return (await provider.getCode(address)) === "0x"
+}
+
+const isOrchestrator = (item) => {
+    return item.id === item.delegate.id;
+}
+
+const filterAddresses = async (arr) => {
+    // remove orchestrators
+    const noOrchs = arr.filter(item => !isOrchestrator(item))
+
+    // remove contract accounts
+    const results = await Promise.all(
+        noOrchs.map(item => isEOA(item.id))
```

Comment:
Doesn't look like `.map()` would return an array of Promises which is what `Promise.all()`.`isEOA()` returns a bool and not a Promise since there there is an await for the `provider.getCode()` call.

Reply:
All async functions always return a promise. I did test it with some dummy values and it worked fine. 

<img width="735" alt="Screenshot 2022-01-06 at 1 57 07 AM" src="https://user-images.githubusercontent.com/23727056/148284423-1b33d30c-5eaa-49ad-9fc7-7c5451f3d66f.png">

<img width="844" alt="Screenshot 2022-01-06 at 1 59 48 AM" src="https://user-images.githubusercontent.com/23727056/148284806-e8fd6368-b751-49f9-b93b-29e7fcb55eb8.png">

Reply:
Oh! Did not know that thanks for the tip. Looks like async functions always wrap the return value in a Promise if it is not already a Promise.